qemu/tests/netuser_buffer_test.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import os
+import time
+
+from avocado.utils import process
+from virttest import error_context
+from virttest import data_dir
+from virttest import utils_net
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Boot guest with "-net user" and virtio-net backend
+    2) Set MTU value in guest
+    3) Compile the script and execute
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    login_timeout = params.get_numeric("login_timeout", 360)
+    vm.wait_for_login(timeout=login_timeout)
+    test_path = os.path.join(data_dir.get_deps_dir(), "netuser/exp.c")
+    vm.copy_files_to(test_path, '~')
+    vm.destroy()
+
+    params["nettype"] = "user"
+    params["vhost"] = ""
+    params = params.object_params(vm.name)
+    vm.create(params=params)
+    serial_session = vm.wait_for_serial_login(timeout=login_timeout)
+
+    def mtu_test():
+        logging.info("Set mtu value and verfied")
+        serial_session.cmd(params["fw_stop_cmd"], ignore_all_errors=True)
+        guest_ifname = utils_net.get_linux_ifname(serial_session,
+                                                  vm.get_mac_address(0))
+        if guest_ifname != 'eth0':
+            test.cancel("Guest device name is not expected")
+        serial_session.cmd(params["set_mtu_cmd"] % guest_ifname)
+        output = serial_session.cmd_output(params["check_mtu_cmd"] %
+                                           guest_ifname)
+        match_string = "mtu %s" % params["mtu_value"]
+        if match_string not in output:
+            test.fail("Guest mtu is not the expected value %s" %
+                      params["mtu_value"])
+
+    def pkg_buffer_test():
+        logging.info("Compile the script and execute")
+        serial_session.cmd("cc -xc -o ~/exp ~/exp.c")
```

Comment:
`cc` should be `gcc`? And recommend define a variable for `exp` and `exp.c`, since you reference them several times, like `test_bin = exp`; and `test_source = exp.c`.

Reply:
Yes, according to the README.md file, gcc should be used. My usage comes from the steps to reproduce the bug. I think both usage should be allowed. Another, is it not intuitive enough to use variable names instead of exp and exp.c? If my understanding is wrong, you can correct me

Reply:
> Yes, according to the README.md file, gcc should be used. My usage comes from the steps to reproduce the bug.

I just simply compare your steps with manual test case description, we'd better keep the automation implementations exactly same as manual test case description, to not make other uses confused.

> I think both usage should be allowed. Another, is it not intuitive enough to use variable names instead of exp and exp.c? If my understanding is wrong, you can correct me

I suppose for maintenance friendly code it is better to define a variable for it when you reference a constant several times. But I am okay if you prefer to keep it this way, since there is lower chance that you would modify the name of the application(One of the advantage to define a variable is that when you need to modify the value you will only modify it at that one place you define it, therefore you will not miss any other where).

Reply:
okay, I have changed to gcc. (exp and exp.c) I want to keep the current usage.